planner/core/logical_plan_builder.go

```diff
@@ -296,11 +296,15 @@ func (b *PlanBuilder) buildJoin(joinNode *ast.Join) (LogicalPlan, error) {
 	joinPlan.SetSchema(expression.MergeSchema(leftPlan.Schema(), rightPlan.Schema()))
 
 	// Set join type.
+	// For outer join case, we need reset not null flag since it column after outer join
+	// could be null.
 	switch joinNode.Tp {
 	case ast.LeftJoin:
 		joinPlan.JoinType = LeftOuterJoin
+		b.isOuterJoin = true
```

Comment:
1. `b.isOuterJoin` has not be set to the original value
2. we don't need to store this variable. We can just remove the `NotNull` flag of all the columns from the inner table in the join result.

Reply:
Maybe we cannot remove the NotNull flag of a column,
since the `*FieldType` of the Column maybe also referred by the `infoSchema`.

Reply:
If `isOuterJoin` is true, and the `col` of `ifnull(col, 0)` is from the outer table,
can we eliminate `ifnull`?

Reply:
> If isOuterJoin is true, and the col of ifnull(col, 0) is from the outer table,
can we eliminate ifnull?

For this case, we cannot eliminate ifnull.

Reply:
why we cannot eliminate for the outer columns?

Reply:
outer join may add null value even if such column having a not null flag.

Reply:
My bad. I mean adding null value for inner table part.